### ISSUE TITLE
Chez test failure on `release` branch

### ISSUE THREAD

**Issue by LiberalArtist** [racket/racket#4359]
I've gotten a new Chez Scheme test failure while updating my Guix patch series to build fb526d1d31f5c26a606b62e2074566c7f5a13e54 from the `release` branch. The full build log is at [z5b0b77px8iscdmnma5q0zgvx5gfli-chez-scheme-for-racket-9.5.9.2.drv.gz](https://github.com/racket/racket/files/9117071/z5b0b77px8iscdmnma5q0zgvx5gfli-chez-scheme-for-racket-9.5.9.2.drv.gz); here's the tail:
```
Hunk #9 succeeded at 8055 (offset 23 lines).
Hunk #10 succeeded at 9331 (offset 23 lines).
Hunk #11 succeeded at 10855 (offset 23 lines).
running ta6le/bin/ta6le/scheme to build all with output to ta6le/mats/output-interpret-0-f-t-f-rmg2/make.out
patch ta6le/mats/experr-interpret-3-f-t-f ../ChezScheme/mats/patch-interpret-3-f-t-f
running ta6le/bin/ta6le/scheme to build all with output to ta6le/mats/output-interpret-3-f-t-f-rmg2/make.out
running ta6le/bin/ta6le/scheme to build all with output to ta6le/mats/output-compile-0-f-f-f-hci-cl/make.out
running ta6le/bin/ta6le/scheme to build all with output to ta6le/mats/output-interpret-3-f-f-f-hci-rmg/make.out
-------- o=0 eoc=f simple --------
-------- o=3 eoc=f simple --------
7d6
< misc.mo:Bug in mat compute-size-increments clause 17 at line 1102, char 3 of /tmp/guix-build-chez-scheme-for-racket-9.5.9.2.drv-0/source/racket/src/ChezScheme/mats/misc.ms
-------- o=0 cp0=t cl=3 cl3 --------
-------- o=3 cp0=t cl=3 cl3 --------
7d6
< misc.mo:Bug in mat compute-size-increments clause 17 at line 1102, char 3 of /tmp/guix-build-chez-scheme-for-racket-9.5.9.2.drv-0/source/racket/src/ChezScheme/mats/misc.ms
-------- o=0 spi=t p=t rmg=2 spi-rmg --------
-------- o=3 spi=t p=t rmg=2 spi-rmg --------
7d6
< misc.mo:Bug in mat compute-size-increments clause 17 at line 1102, char 3 of /tmp/guix-build-chez-scheme-for-racket-9.5.9.2.drv-0/source/racket/src/ChezScheme/mats/misc.ms
-------- o=0 eval=interpret cl=6 cl6 --------
-------- o=3 eval=interpret cl=6 cl6 --------
-------- o=0 cp0=t eval=interpret rmg=2 rmg2 --------
-------- o=3 cp0=t eval=interpret rmg=2 rmg2 --------
-------- o=0 hci=101 eoc=f cl=9 hci-cl --------
-------- o=3 hci=101 eval=interpret rmg=2 hci-rmg --------
some output differs from expected
 in build-one
 in loop
 in module->hash
make: *** [Makefile:30: test] Error 1

Test suite failed, dumping logs.
error: in phase 'check': uncaught exception:
%exception #<&invoke-error program: "make" arguments: ("test" "-j" "12" "ZUO=/gnu/store/06y37sa4d4d26dsrc0ikk352s2i2jdjb-zuo-1.0-racket8.5.900-guix1/bin/zuo" "STEXLIB=/gnu/store/wsmsnsqlm8py8zzvijjxr75593g949rq-stex-1.2.2-3.afa6075/lib/stex") exit-status: 2 term-signal: #f stop-signal: #f> 
phase `check' failed after 3766.6 seconds
command "make" "test" "-j" "12" "ZUO=/gnu/store/06y37sa4d4d26dsrc0ikk352s2i2jdjb-zuo-1.0-racket8.5.900-guix1/bin/zuo" "STEXLIB=/gnu/store/wsmsnsqlm8py8zzvijjxr75593g949rq-stex-1.2.2-3.afa6075/lib/stex" failed with status 2
```
I don't yet understand what's happening, but I've saved the build directory if there's anything that would be useful.

**Comment by mflatt** [comment]
I think the problem here is that the test is broken. The test is meant to call `compute-size-increments` with only a single thread active, but it's possible that a sleep is misinterpreted as a thread being blocked, and in that case the thread can become unblocked before `compute-size-increments` is reached. The failures are probably all with `o=3` because that ends up with just the right thread scheduling to provoke a problem; on my machine, I provoked failures by adding random spinning in the created thread.

Meanwhile, the `compute-size-increments` function should be documented to require that it's called in the only active thread, and it should check that requirement. So, I'll make that change along with fixing the test.

**Comment by LiberalArtist** [comment]
I've cherry-picked 153ff9acb7ad63717a50bb26cd5aaa053870c666 as e673c71 on top of f90bcc15b87e16651c67404a3e3231c366f712bc from the `release` branch, and I can confirm that it fixes the test failure. Is it possible to get this in 8.6? (cc @jbclements) I guess it's debatable whether it's technically a "regression" if the test was already broken, but at least on my machine I'd never hit this failure before.

**Comment by LiberalArtist** [comment]
I'm sorry to report that my latest build has hit a variety of new errors: the log is at 
[gfpb5mlpv8cdirfhaz4m8xqbhxvhvi-chez-scheme-for-racket-9.5.9.2.drv.gz](https://github.com/racket/racket/files/9175070/gfpb5mlpv8cdirfhaz4m8xqbhxvhvi-chez-scheme-for-racket-9.5.9.2.drv.gz), which is from building https://github.com/LiberalArtist/racket/commit/d1f5d402c9f850cb8dcc0039b7562a888ce504f1.

**Comment by mflatt** [comment]
The diff in that output is showing only `Expected error` lines. Also, the direction of the diff is `>`, which means lines in the second argument to `diff` that are not in the first argument. The first argument to `diff` in `mats/build.zuo` is the expected-error file, while the second argument to `diff` is output from the test run. Based on the line numbers and that `o=0 eoc=f simple` and `o=0 hci=101 eoc=f cl=9 hci-cl` use that same `experr-compile-0-f-f-f`, it seems that the test-output file for `o=0 eoc=f simple` has an extra copy of lines 10615 through 10898 inserted at line 10642. But there I get stuck, because I don't how that might happen.

**Comment by LiberalArtist** [comment]
I tried enabling Chez tests again, and they're passing now with https://github.com/LiberalArtist/racket/commit/d52bbde2ae635d944faea56b36efaaff9c87775e. I have no idea why they're passing, but I have no idea why they weren't working before, either, so I guess I'll close this. (CC @jbclements, just in case you were watching this.)

**Comment by LiberalArtist** [comment]
It happened again! This time I was building https://github.com/LiberalArtist/racket/commit/477d6f9d2872d4fe3839693f14594960110f1719: the build log is 
[5gj58rk5drcfgfvskx2kq1zcclcdpjad-chez-scheme-for-racket-9.5.9.2.drv.gz](https://github.com/racket/racket/files/9239273/5gj58rk5drcfgfvskx2kq1zcclcdpjad-chez-scheme-for-racket-9.5.9.2.drv.gz). I'll see if I can reproduce the failure consistently.

**Comment by mflatt** [comment]
In that log, it looks like all the tests fail because "primvars.ms" isn't found. Something going wrong with paths?

**Comment by LiberalArtist** [comment]
I think you're right that it's a problem with paths. I remembered that I had switched to an out-of-source build for Racket's Chez sometime since 8.5. Just now, I was able to build Chez from c3dd01055ed7589a18136904510fe4db557d6e77 (`master`) in-source successfully with tests passing, whereas an otherwise-identical out-of-source build had test failures. IIRC, building out-of-source had been working in <https://racket.discourse.group/t/1063/37>, though.

I'll next try an in-source build with https://github.com/racket/racket/commit/153ff9acb7ad63717a50bb26cd5aaa053870c666 cherry-picked on top of 8.6: if that works, I can try to debug the out-of-source build after getting the 8.6 release into Guix.

**Comment by LiberalArtist** [comment]
> I'll next try an in-source build with [153ff9a](https://github.com/racket/racket/commit/153ff9acb7ad63717a50bb26cd5aaa053870c666) cherry-picked on top of 8.6: if that works, I can try to debug the out-of-source build after getting the 8.6 release into Guix.

Unfortunately, that didn't work. Still, it seems better to just skip these tests in Guix for 8.6 and figure out the problem on master, rather than piling cherry-pick on top of cherry-pick.

**Comment by LiberalArtist** [comment]
This seems to be fixed in 8.7.
